Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,90 @@
+require "set"
+
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.10.1.tar.gz"
+  sha256 "e571382dbccca0c37f38d097a12bf54874190a312e55ce22534cc6b1ddb8ffaf"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  WITHOUT_OPTIONS = {
+    "compute" => ["ec2", "elasticbeanstalk", "lambda"],
+    "storage" => ["s3", "cloudfront", "elasticfilesystem", "glacier", "importexport", "storagegateway"],
+    "database" => ["rds", "dynamodb", "elasticache", "redshift", "dms", "sdb"],
+    "networking" => ["directconnect", "route53", "route53domains"],
+    "devtools" => ["codecommit", "codedeploy", "codepipeline"],
+    "management" => ["cloudformation", "cloudtrail", "config", "opsworks", "ecr", "ecs", "autoscaling", "support", "logs", "monitoring", "ssm"],
+    "security" => ["identity-management", "access-management", "inspector", "waf", "iam", "cloudhsm", "acm", "ds", "kms", "sts"],
+    "analytics" => ["datapipeline", "kinesis", "firehose", "marketplacecommerceanalytics"],
+    "mobile" => ["cognito-identity", "cognito-sync", "devicefarm", "mobileanalytics", "sns"],
+    "appservices" => ["apigateway", "cloudsearch", "cloudsearchdomain", "elastictranscoder", "sqs", "swf", "email"],
+    "other" => ["elasticloadbalancing", "elasticmapreduce", "es", "events", "machinelearning", "queues", "transfer", "iot", "gamelift", "workspaces", "meteringmarketplace"],
+  }.freeze
+
+  WITHOUT_OPTIONS.each { |category, clients| option "without-#{category}", "Exclude clients: #{clients.join(", ")}" }
```

Comment:
Having lots of options is bad feng shui; you'll note that we don't have e.g. a way to build only certain Boost modules. This looks like something that will change as Amazon rolls out new services so it adds more work for us. --without options are additionally troubling in part because Homebrew does not have a way to express a dependency on a package that did not have a recommended feature disabled.

It sounds like you have unusual needs; one way you could use Homebrew to install this software is to [maintain your own tap](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap). The linked document has some advice and we would be happy to help.

Alternatively, Mike's suggestion to send a PR upstream to allow building both dynamic & static libraries at the same time sounds great to me.